grpcclient/lc_client.go

```diff
@@ -84,6 +82,12 @@ type LcClientIf interface {
 	GetFile(ctx context.Context, key []byte, filePath string) (*immuschema.Entry, error)
 
 	Connect() (err error)
+	IsConnected() bool
+
+	ConsistencyCheck(ctx context.Context) error
+
+	GetApiKey() string
```

Comment:
Api key need to be protected by mutex now. Need to be thread safe since this value is modified in case we provide api-key using context

Reply:
I will also remove the exposure of this field but I need to do more work since on CAS server is still used vcn(old version not enterprise) that rely on api-key info. So it's required to switch from vcn to cas package and  the update of cas itself with new  sdk